.circleci/config.yml

```diff
@@ -1,13 +1,70 @@
-# Use the latest 2.1 version of CircleCI pipeline process engine. See: https://circleci.com/docs/2.0/configuration-reference
 version: 2.1
-# Use a package of configuration called an orb.
-orbs:
-  # Declare a dependency on the welcome-orb
-  welcome: circleci/welcome-orb@0.4.1
-# Orchestrate or schedule a set of jobs
+
+definitions:
+  release-tag: &release-tag /^[0-9]+\.[0-9]+\.[0-9]+$/
+  image-name: &image-name cobli/helm-client
+  helm-version-matrix: &helm-version-matrix
+    matrix:
+      parameters:
+        helm-version: 
+          - "3.2.1"
+
 workflows:
-  # Name the workflow "welcome"
-  welcome:
-    # Run the welcome/run job in its own container
+  build-and-publish-docker-image:
     jobs:
-      - welcome/run
\ No newline at end of file
+      - docker/hadolint
+      - build-and-push:
+          <<: *helm-version-matrix
+          push: false
+          filters:
+            branches:
+              ignore: /^master$/
+            tags:
+              ignore: *release-tag
+          requires:
+            - docker/hadolint
+      - build-and-push:
+          <<: *helm-version-matrix
+          context: coblicd-dockerhub-login
+          push: true
+          filters:
+            branches:
+              only: /^master$/
+            tags:
+              only: *release-tag
+          requires:
+            - docker/hadolint
```

Comment:
Shouldn't we create a job for hadolint? or pass these parameters?
```
      - docker/hadolint:
          dockerfile: path/to/Dockerfile
          ignore-rules: 'DL4005,DL3008'
          trusted-registries: 'docker.io,my-company.com:5000'
```
ref.: https://circleci.com/orbs/registry/orb/circleci/docker#usage-hadolint

Reply:
We are calling hadolint job of this orb with default parameters (Dockerfile on root, not ignoring any rules, and trusting on docker.io).